### DISCUSSION TITLE
Font combination suggestions?

### DISCUSSION THREAD

**Discussion by ldatura** [kovidgoyal/kitty#3445] [Q&A]
Hi! I love the speed and performance of kitty… there's nothing better out there! However, I'm still looking for a combination of fonts that will look good, so I'd like to ask for a few suggestions. Since kitty font requirements are so strict, the ones I've tried usually will have problems with italics (crop or reduce the sizes of j, f, Q, M, %, and so on). Is there any font that works fine that I could try?

**Comment by kovidgoyal** [comment]
```
font_family Operator Mono Book
bold_font Operator Mono Medium
italic_font Operator Mono Book Italic
bold_italic_font Operator Mono Medium Italic

symbol_map U+E0A0-U+E0A2,U+E0B0-U+E0B3 PowerlineSymbols
# Rule for / which is much wider in the italic Operator Mono variant, leading to truncated rendering
symbol_map U+002f Operator Mono Book
```

**Reply by ldatura** [reply]
Thanks for the suggestion! Still, the slash symbol with this font needs a workaround and there's also a problem with X being truncated when italics or bold italics!

**Comment by kovidgoyal** [comment]
The workaround is one line in kitty.conf and I dont see the X being
truncated in italics at least, dont really use bold-italics. And if it
really bothers you, use adjust_cell_width to make the cells wide enough
to fit the characters from whatever font you prefer.

On Tue, Apr 06, 2021 at 08:43:57AM -0700, Caio Prado wrote:
> Thanks for the suggestion! Still, the slash symbol with this font needs a workaround and there's also a problem with X being truncated when italics or bold italics!
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/3445#discussioncomment-575581

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by ldatura** [reply]
I understand the workaround is simple, but it still implies using a glyph from a different typeface! As for the 'X', even in italics it does happen, though it's just a little bit. The problem can be mitigated with adjust_column_width as you suggested.
![test](https://user-images.githubusercontent.com/910629/113800010-c61bf700-9788-11eb-89ba-d25e2e00eb96.png)

I am aware of these adjustments as they have been suggested several times in issues for several different fonts. I was just wondering if there is any font selection that would just work as the font was intended to without resorting to hide rendering problems every here and there…

**Comment by trygveaa** [comment]
I use [JetBrains Mono](https://www.jetbrains.com/lp/mono/). It supports italics, bold etc. and works great with kitty.

**Reply by ldatura** [reply]
Thanks for the suggestion… I’ve tried using JetBrains Mono, but unfortunately some glyphs are still very badly rendered in italics (such as %).

**Reply by trygveaa** [reply]
It's not badly rendered for me, though apparently % look pretty similar in italics as in normal.

**Comment by EricDriussi** [comment]
I think a setting to just let the glyphs bleed into the prev or following cell would be greatly appreciated (I for one would love that).
I really love kitty but these kind of strict requirements are driving me away from it.

My job consists of reading code for hours on end. 
I'm using whatever font is easy on my eyes, not the one that fits my emulator requirements.

Sure there are workarounds, but they could all be avoided by just letting the user choose if they want that behavior or not.

You state in the docs that letting the character overflow into neighboring cells can look terrible if it that cell isn't empty.
With all due respect, I don't see how that's for anyone to decide but the user. Most other emulators behave like this and their users's eyes are not bleeding out as far as I can tell.
There are however quite a few issues and posts regarding characters being truncated on kitty...

I really hope that this doesn't come off as entitled. I really do appreciate your work.
I just hope I could get to chose these sort of behaviors to be on or off.
